src/Magnum/Trade/MeshData.cpp

```diff
@@ -0,0 +1,150 @@
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "MeshData.h"
+
+namespace Magnum { namespace Trade {
+
+MeshData::MeshData(Containers::Array<char>&& indexData, Containers::Array<char>&& vertexData, const MeshIndexData indices, Containers::Array<MeshAttributeData>&& attributes, const void* const importerState) noexcept: _indexData{std::move(indexData)}, _vertexData{std::move(vertexData)}, _indices{indices}, _attributes{std::move(attributes)}, _importerState{importerState} {
+    /** @todo assert that index data are divisible by type size, same for other types */
+    /** @todo assert that there's at least one position? */
+    /** @todo assert that all arrays have the same amount of elemenc */
+}
+
+MeshData::~MeshData() = default;
+
+MeshData::MeshData(MeshData&&) noexcept = default;
+
+MeshData& MeshData::operator=(MeshData&&) noexcept = default;
+
+UnsignedInt MeshData::indexCount() const {
+    switch(_indices.type) {
+        case MeshIndexType::UnsignedByte: return _indices.data.size();
+        case MeshIndexType::UnsignedShort: return _indices.data.size()/2;
+        case MeshIndexType::UnsignedInt: return _indices.data.size()/4;
+    }
+
+    CORRADE_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+}
+
+UnsignedInt MeshData::attributeCount(const MeshAttributeName name) const {
+    UnsignedInt count = 0;
+    for(const MeshAttributeData& attribute: _attributes)
+        if(attribute.name == name) ++count;
+    return count;
+}
+
+MeshAttributeName MeshData::attributeName(UnsignedInt id) const {
+    CORRADE_ASSERT(id < _attributes.size(),
+        "Trade::MeshData::attributeName(): index" << id << "out of range for" << _attributes.size() << "attributes", {});
+    return _attributes[id].name;
+}
+
+MeshAttributeType MeshData::attributeType(UnsignedInt id) const {
+    CORRADE_ASSERT(id < _attributes.size(),
+        "Trade::MeshData::attributeName(): index" << id << "out of range for" << _attributes.size() << "attributes", {});
+    return _attributes[id].type;
+}
+
+UnsignedInt MeshData::attributeFor(const MeshAttributeName name, UnsignedInt id) const {
+    for(std::size_t i = 0; i != _attributes.size(); ++i)
+        if(_attributes[i].name == name && !id--) return i;
+    CORRADE_ASSERT(false, "Trade::MeshData: index" << id << "out of bounds for attribute" << name, {});
+}
+
+namespace {
+    template<class T> Containers::Array<UnsignedInt> convertIndices(const Containers::ArrayView<const char> data) {
+        const auto input = Containers::arrayCast<const T>(data);
+        Containers::Array<UnsignedInt> output{input.size()};
+        for(std::size_t i = 0; i != input.size(); ++i) output[i] = input[i];
+        return output;
+    }
+}
+
+Containers::Array<UnsignedInt> MeshData::indices() const {
```

Comment:
I wonder if there could be additional API to pass the index Data as is without conversion. I think GL doesn't need Integers, right? (Ah, I see `releaseIndices` is just the thing 👍 ) 

That would be a first step to allowing the loading of pre-`MeshTools::compile(...)`-ed meshes :)

Reply:
`indices<T>()` in the header is what you're looking for -- and together with the equivalent functionality for attribute data this is the *main* reason why I'm doing this rework, otherwise zero-copy import wouldn't be possible ;) 

This is just a convenience function for cases where you don't want to handle all three types.